metrics/providers/micrometer/src/main/java/io/helidon/metrics/providers/micrometer/MMeterRegistry.java

```diff
@@ -215,39 +221,65 @@ static MMeterRegistry applyMetersProvidersToRegistry(MetricsFactory factory,
 
     @Override
     public void close() {
-        onAddListeners.clear();
-        onRemoveListeners.clear();
-        List.copyOf(meters.values()).forEach(this::remove);
-        meters.clear();
-        buildersByPromMeterId.clear();
-        scopeMembership.clear();
-        metersById.clear();
+        lock.writeLock().lock();
+        try {
+            onAddListeners.clear();
+            onRemoveListeners.clear();
+            List.copyOf(meters.values()).forEach(this::remove);
+            meters.clear();
+            buildersByPromMeterId.clear();
+            scopeMembership.clear();
+            metersById.clear();
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     @Override
     public List<io.helidon.metrics.api.Meter> meters() {
-        return meters.values()
-                .stream()
+        lock.readLock().lock();
+        List<io.helidon.metrics.api.Meter> temp;
+        try {
+            temp = new ArrayList<>(meters.values());
+        } finally {
+            lock.readLock().unlock();
+        }
+        return temp.stream()
                 .map(io.helidon.metrics.api.Meter.class::cast)
                 .toList();
     }
 
     @Override
     public Collection<io.helidon.metrics.api.Meter> meters(Predicate<io.helidon.metrics.api.Meter> filter) {
-        return meters.values()
-                .stream()
+        lock.readLock().lock();
+        List<io.helidon.metrics.api.Meter> temp;
+        try {
+            temp = new ArrayList<>(meters.values());
```

Comment:
Hmm; this idiom with the streaming below it seems verbose. What about:
```
return List.copyOf(meters.values()).stream()
    .map(...)
    ....
```
…all here under lock instead?

Reply:
Copying the array inside the lock but doing the rest of the work outside lets us release the lock sooner.